src/actions/tree-actions.js

```diff
@@ -0,0 +1,71 @@
+import * as types from './action-types';
+
+export const updateTree = (tree) => {
+  return {
+    type: types.UPDATE_TREE,
+    tree
+  };
+};
+
+export const regressPath = (amt) => {
+  return {
+    type: types.REGRESS_PATH,
+    amt
+  };
+};
+
+export const advancePath = (pro, index) => {
+  return {
+    type: types.ADVANCE_PATH,
+    pro,
+    index
+  };
+};
+
+// export const setConfidence = (statement, confidence) => {
```

Comment:
Are, along with all the below comment functions, commented out because they're unneeded, or they simply don't work? From what I understand you're updating the **entire** tree every time a single piece of data is changed. Please correct me if I'm wrong.